src/spatialdata/_core/query/relational_query.py

```diff
@@ -730,23 +751,19 @@ def _locate_value(
             region = table.uns[TableModel.ATTRS_KEY][TableModel.REGION_KEY]
             if element_name in region:
                 # check if the value_key is in the table
-                if value_key in table.obs.columns:
-                    value = table.obs[value_key]
-                    is_categorical = pd.api.types.is_categorical_dtype(value)
-                    origins.append(_ValueOrigin(origin="obs", is_categorical=is_categorical, value_key=value_key))
-                # check if the value_key is in the var
-                elif value_key in table.var_names:
-                    origins.append(_ValueOrigin(origin="var", is_categorical=False, value_key=value_key))
+                origins = _get_table_origins(element=table, value_key=value_key, origins=origins)
+
     return origins
 
 
 def get_values(
     value_key: str | list[str],
-    element: SpatialElement | None = None,
+    element: SpatialElement | AnnData | None = None,
     sdata: SpatialData | None = None,
     element_name: str | None = None,
     table_name: str | None = None,
-) -> pd.DataFrame:
+    return_obsm_as_is: bool = False,
+) -> pd.DataFrame | npt.NDArray[Any]:
```

Comment:
I will replace this with `ArrayLike` from `from spatialdata._types import ArrayLike` in the next commit.